au4/src/projectscene/view/playcursor/playpositionactioncontroller.cpp

```diff
@@ -60,26 +60,63 @@ void PlayPositionActionController::playPositionIncrease()
 
 void PlayPositionActionController::init()
 {
+    IProjectViewStatePtr viewState = globalContext()->currentProject()->viewState();
+
+    viewState->isSnapEnabled().ch.onReceive(this, [this](bool snapEnabled){
+        if (snapEnabled) {
+            snapCurrentPosition();
+        }
+    });
+
+    viewState->snapType().ch.onReceive(this, [this, viewState](SnapType snapType){
+        Q_UNUSED(snapType);
+        if (viewState->isSnapEnabled().val) {
+            snapCurrentPosition();
+        }
+    });
+
+    viewState->isSnapTripletsEnabled().ch.onReceive(this, [this, viewState](bool tripletsEnabled){
+        Q_UNUSED(tripletsEnabled);
+        if (viewState->isSnapEnabled().val) {
+            snapCurrentPosition();
+        }
+    });
+
     dispatcher()->reg(this, PLAY_POSITION_DECREASE, this, &PlayPositionActionController::playPositionDecrease);
     dispatcher()->reg(this, PLAY_POSITION_INCREASE, this, &PlayPositionActionController::playPositionIncrease);
 }
 
+void PlayPositionActionController::snapCurrentPosition()
+{
+    audio::secs_t currentPlaybackPosition = globalContext()->playbackState()->playbackPosition();
+    double newXPosition = m_context->timeToPosition(currentPlaybackPosition);
+    audio::secs_t secs = m_context->positionToTime(newXPosition, true);
+    if (muse::RealIsEqualOrMore(secs, 0.0)) {
+        dispatcher()->dispatch("playback_seek", ActionData::make_arg1<double>(secs));
+    }
+}
+
 void PlayPositionActionController::applySingleStep(Direction direction)
 {
     IProjectViewStatePtr viewState = globalContext()->currentProject()->viewState();
     bool snapEnabled = viewState->isSnapEnabled().val;
 
     audio::secs_t currentPlaybackPosition = globalContext()->playbackState()->playbackPosition();
-    double newXPosition;
-    if (direction == Direction::Left) {
-        newXPosition = m_context->timeToPosition(currentPlaybackPosition) - 1;
+    audio::secs_t secs = 0;
+
+    if (snapEnabled) {
+        double newXPosition = m_context->timeToPosition(currentPlaybackPosition);
+        secs = m_context->singleStepToTime(newXPosition, direction);
     } else {
-        newXPosition = m_context->timeToPosition(currentPlaybackPosition) + 1;
-    }
+        double newXPosition;
```

Comment:
please initialize with default value (code may change over time... at some point the value of this variable may not be assigned)

Reply:
fixed